pkg/stanza/fileconsumer/matcher/internal/filter/sort.go

```diff
@@ -174,3 +174,32 @@ func (m mtimeSortOption) apply(items []*item) ([]*item, error) {
 func SortMtime() Option {
 	return mtimeSortOption{}
 }
+
+type excludeOlderThanOption struct {
+	minMtime time.Time
+}
+
+func (eot excludeOlderThanOption) apply(items []*item) ([]*item, error) {
+	filteredItems := make([]*item, 0, len(items))
+	var errs error
+	for _, item := range items {
+		path := item.value
+		fi, err := os.Stat(path)
+		if err != nil {
+			errs = multierr.Append(errs, err)
+			continue
+		}
+
+		if !fi.ModTime().Before(eot.minMtime) {
+			filteredItems = append(filteredItems, item)
+		}
+	}
+
+	return filteredItems, errs
+}
+
+// ExcludeOlderThan excludes files whose modification time is older than the specified age.
+func ExcludeOlderThan(age time.Duration) Option {
+	minMtime := time.Now().Add(-age)
```

Comment:
Based on how we use this, we're setting a fixed time at startup and then never updating it. I would have assumed we want to determine the max time when `Match` is called. WDYT?

Reply:
Ah yes, I agree.  The previous iteration of this implementation took a `time.Time` as input to this function so setting that at startup made sense.  But with a `time.Duration` that doesn't really make sense.

Reply:
Fixed in 940243861b.